### PR TITLE
[CM-0] Refactor consent manager

### PR DESCRIPTION
**Pull request by ganesanarun**
* Never throw exception, always use Mono.error
* Return Flux from repository, and do collect in service when you need it
* Don't return Flux and break everything, if you know that you are supposed to return Mono
* Make sure you break after 120 character per line, readable horizontally too
* Don't use magic strings in tests

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637447&bid=18462651)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637447&bid=18462651)

File: src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java [hunk 1]
```diff
@@ -84,10 +84,11 @@ public IdentityServiceClient(WebClient.Builder webClientBuilder,
                 .retrieve()
                 .onStatus(HttpStatus::is4xxClientError, clientResponse -> Mono.error(userNotFound()))
                 .onStatus(HttpStatus::isError, clientResponse -> Mono.error(networkServiceCallFailed()))
-                .bodyToFlux(KeyCloakUserRepresentation.class);
+                .bodyToFlux(KeyCloakUserRepresentation.class)
```
Comment (dheeraj-p) [inline src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java:87]: Any specific reason for not using `bodyToMono`?

File: src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java [hunk 2]
```diff
@@ -97,7 +98,8 @@ public IdentityServiceClient(WebClient.Builder webClientBuilder,
                 .retrieve()
                 .onStatus(HttpStatus::is4xxClientError, clientResponse -> Mono.error(userNotFound()))
                 .onStatus(HttpStatus::isError, clientResponse -> Mono.error(networkServiceCallFailed()))
-                .bodyToFlux(KeyCloakUserCredentialRepresentation.class);
+                .bodyToFlux(KeyCloakUserCredentialRepresentation.class)
```
Comment (dheeraj-p) [inline src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java:101]: Same as above

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones removed
==============
+ src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  -7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637447&bid=18462651)

**Pull request merged**